### ISSUE TITLE
ENH: SuperLU and distributed systems

### ISSUE THREAD

**Issue by GregStroot** [scipy/scipy#16112]
### Is your feature request related to a problem? Please describe.

Yes. When using SuperLU on distributed systems, superLU is not able to store the decomposition in a distributed fashion.

E.g. if I have two nodes with 128 gb each and my LU decomposition uses 180gb, the superLU call will fail due to lack of memory. I believe this is because it tries to store the 180gb block on a single node, which only has 128gb of memory

If it could be store across different nodes then we could better utilize the resources we are using instead of wasting memory 

### Describe the solution you'd like.

I (perhaps naively) think this could be solved by adding an option to use SuperLU_dist. The issue that I potentially see is that when one calls SuperLU.solve(), it will need to use MPI to communicate between the shared memory location (though this may already be handled in SuperLU_dist?

### Describe alternatives you've considered.

The only other alternative I see would be save my matricies (generated within python) that I want to run my algorithm on and then run SuperLU (or MUMPS or likewise) natively in another language. Without  rewriting SuperLU as a whole and implementing MPI there-in, I think the best way to go about this would be to use the packages already created and interface with this framework

**Comment by GregStroot** [comment]
For others that run into a similar issue, I found two potential work-arounds:

1. Work with petsc4py and you may use either MUMPs or SuperLU in parallel. This requires some work, but it is fairly easy to convert algorithms from scipy sparse to petsc. This has the benefit of running the computation in parallel

2. Try the umfpack driver for scipy. One potential reason (I believe) that this issue comes up is that SuperLU uses `int32` for the indicies and if you are using large enough matricies then the computation will fail (even when there is enough memory). Umfpack, in constrast, uses  `int64` so this may work around this issue. I haven't tested this theory, but it was another idea I had of why SuperLU may be failing

**Comment by rgommers** [comment]
Thanks for the report @GregStroot. The workarounds seem useful. I'm afraid this issue is not actionable for SciPy - distributed system support is out of scope. There is indeed a distributed version of SuperLU (`SuperLU_DIST`, see https://portal.nersc.gov/project/sparse/superlu/), but we won't integrate that into SciPy.

If you really do need proper distributed support, you can indeed use `petsc4py`. For dense distributed LU decomposition you can also use [dask.array.linalg.lu](https://docs.dask.org/en/stable/generated/dask.array.linalg.lu.html#dask.array.linalg.lu), but there's no sparse LU for Dask.

I'll close this issue as wontfix. Please feel free to continue the conversation, in case there's more alternatives or things to discuss.
